x/lockup/spec/02_state.md

```diff
@@ -0,0 +1,49 @@
+<!--
+order: 2
+-->
+
+# State
+
+## Locked coins management
+
+Locked coins are all stored in module account for `lockup` module which is called `LockPool`.
+When user lock coins within `lockup` module, it's moved from user account to `LockPool` and a record (`PeriodLock` struct) is created.
+
+Once the period is over, user can withdraw it at anytime from `LockPool`.
+Note:
+- Do we need automate withdraw from the pool?
```

Comment:
No need imo

Reply:
Although to be fair, by that logic, we could get rid of the unbonding queue in staking as well 🤔 

We even suggested that here (https://github.com/cosmos/cosmos-sdk/issues/992)  but then never did it lol...  If I had to guess probably a combination of:
1. Redelegations would get more messy
2. UX reasons

Reply:
We even suggested that here (https://github.com/cosmos/cosmos-sdk/issues/992)  but then never did it lol...

If I had to guess, probably a combination of:
1. Redelegations would get more messy
2. UX reasons

Reply:
Given that #1 isn't an issue for this, let's not automate for now.

Reply:
ok